From b007e51d7d776d7f31e7e300b7af5f56cfac18c4 Mon Sep 17 00:00:00 2001
From: Roger Pau Monne <roger.pau@citrix.com>
Date: Thu, 13 Sep 2012 15:18:59 +0100
Subject: [PATCH] setup-xen-dom0: update to upcoming 4.2

New version of Xen (4.2) doesn't require udev, so update the
script to reflect those changes. Also, only include a module
in /etc/modules if we can load it successfully.
---
 setup-xen-dom0.in | 20 +++++++-------------
 1 file changed, 7 insertions(+), 13 deletions(-)

diff --git a/setup-xen-dom0.in b/setup-xen-dom0.in
index a9c993a..0e7b3b3 100644
--- a/setup-xen-dom0.in
+++ b/setup-xen-dom0.in
@@ -3,26 +3,20 @@
 # simple script to setup basic xen dom0
 
 # kernel modules
-for mod in xen_netback xen_blkback xenfs xen-platform-pci xen_wdt tun; do
-	modprobe $mod
-	grep -q -w $mod /etc/modules || echo $mod >> /etc/modules
+for mod in xen_netback xen_blkback xenfs xen_pciback xen_wdt tun; do
+	if modprobe $mod; then
+		grep -q -w $mod /etc/modules || echo $mod >> /etc/modules
+	fi
 done
 
 # install packages
-apk add -q xen udev
+apk add -q xen
 
 # setup and start services
-rc-update -q del mdev sysinit
-rc-update -q add udev sysinit
-for svc in udev-postmount xenstored xenconsoled; do
+for svc in xenstored xenconsoled xenqemu; do
 	rc-update -q add $svc default
 done
 
-if ! rc-service -q udev status; then
-	rc-service udev start
-	rc-service udev-postmount start
-fi
-
 rc-service xenstored start
 rc-service xenconsoled start
-
+rc-service xenqemu start
-- 
1.7.12

